library_generation/generate_repo.py

```diff
@@ -39,6 +41,16 @@ def generate_from_yaml(
     If specified with an empty list, then no library will be generated.
     If not specified, all libraries in the configuration yaml will be generated.
     """
+    # Determine whether the generated repo is a proto-only repository.
+    # Using libraries in generation config without filtering out target library
+    # names because the remaining library may not qualify as proto-only
+    # library, e.g., iam.
+    proto_only_repo = False
+    for library in config.libraries:
+        if library.get_library_name() in PROTO_ONLY_LIBRARIES:
```

Comment:
In other words, what's the difference between checking for all libraries being `proto_only` vs having a single `common-protos` in the library list?

Reply:
The `proto-only` libraries means they are dependencies of client libraries (which may or may not have a service defined in proto). Maybe `common_proto_libraries` makes more sense?

When you said proto-only libraries in google-cloud-java, do you mean `java-alloydb-connectors`? I don't think `java-alloydb-connectors` is a proto-only library using the above definition because it is not a dependency of other client libaries.

Reply:
We may not need this parameter.

Reply:
I introduced `contains_common_protos` in generation config object.

Reply:
Thank you. I think this change would make it necessary to change the PR description as well. Maybe to something that explains that this PR enables support for generating sdk-platform-java.